src/Uno.Extensions.Navigation.UI/RouteNotifier.cs

```diff
@@ -53,7 +55,9 @@ public void EndNavigation(INavigator navigator, IRegion region, NavigationReques
 				Logger.LogTraceMessage($"Post-navigation (summary):\n{navigationSegments[id]}");
 			}
 			navigationSegments.Remove(id);
-			RouteChanged?.Invoke(this, new RouteChangedEventArgs(region.Root(), response?.Navigator));
+
+			var myRegion = _initialRegion ?? region;
```

Comment:
I'm assuming here is where we would need to resolve the correct `_initialRegion` based on the current `NavigationRequest` ID